canopen_fake_slaves/include/canopen_fake_slaves/basic_slave.hpp

```diff
@@ -36,11 +36,19 @@ class SimpleSlave : public canopen::BasicSlave
 {
 public:
   using BasicSlave::BasicSlave;
+  ~SimpleSlave()
+  {
+    if (message_thread.joinable())
+    {
+      message_thread.join();
+    }
+  }
 
 protected:
+  std::thread message_thread = nullptr;
```

Comment:
This does not work on humble. Maybe by putting this into `unique_ptr` could be a solution. Or even simpler, but not so nice, adding a new flag variable.